### ISSUE TITLE
different motor board

### ISSUE THREAD

**Issue by melidron** [jamieboyd/MC2100_treadmill#1]
have a treadmill with damaged control board and the motor board pins are like I show in the photo I send... Any idea if your project can be connected and control this board? Thanks in advanced
![12 PINS](https://github.com/user-attachments/assets/de4997f7-55ed-4be1-be5f-e65338870b2a)

**Comment by jamieboyd** [comment]
I don't know for sure. The mc2100 uses PWM control for speed. My code takes button presses for speed+ and speed- and adjusts PWM duty cycle. Plus it has a stop signal for Hall sensor stop button.  A good diagram of the pinout for the mc2100 controller. for reference is [here]( https://dl.dropboxusercontent.com/s/3try02vjpvf8jur/Treadmill%20Schematic.jpg ). You can see it differs from the one you have.

Assuming the bottom connector in your diagram shows the pinout for the cable that connects the control panel to the motor controller, then it appears that you just need to send a high-low pulse on the FAST or SLOW pin to adjust treadmill speed, so need for PWM. Which makes it easier. I presume SPD pins are input for speed. On my treadmill, there was a reed switch that activated with each revolution of the treadmill belt, so could measure the time between transitions on the pin to calculate speed. But I did not get around to putting that in the code. Similarly the incline sensors VR1-3 are input pins and carry a signal related to incline.

**Comment by melidron** [comment]
> I don't know for sure. The mc2100 uses PWM control for speed. My code takes button presses for speed+ and speed- and adjusts PWM duty cycle. Plus it has a stop signal for Hall sensor stop button. A good diagram of the pinout for the mc2100 controller. for reference is [here](https://dl.dropboxusercontent.com/s/3try02vjpvf8jur/Treadmill%20Schematic.jpg). You can see it differs from the one you have.
> 
> Assuming the bottom connector in your diagram shows the pinout for the cable that connects the control panel to the motor controller, then it appears that you just need to send a high-low pulse on the FAST or SLOW pin to adjust treadmill speed, so need for PWM. Which makes it easier. I presume SPD pins are input for speed. On my treadmill, there was a reed switch that activated with each revolution of the treadmill belt, so could measure the time between transitions on the pin to calculate speed. But I did not get around to putting that in the code. Similarly the incline sensors VR1-3 are input pins and carry a signal related to incline.

Look atthe "shorting" of the photo down and right. May help you understand more how my board works... I have no idea how to use the SPD pins or what is the use of them too ... the lines i draw are lines the do connectivity between the pins... I dont know how to make your code work with my board... Any informations u can give me would be very useful because I have it almost 2 years innactive because of damaged control board... Thank you again my friend

![F85board](https://github.com/user-attachments/assets/dc642b00-2198-41a8-8055-b8442043c1b9)

![2024-09-25 231850](https://github.com/user-attachments/assets/ef958943-3857-4418-b517-78d7c694c55e)

**Comment by jamieboyd** [comment]
Yes, the "shorting" diagram indicates that you don't need PWM to do speed control  just a simple switch. For bare bones open loop speed control, maybe you don't even need a microcontroller. If you have the make and model number of the treadmill, perhaps we can find some more information on the signals that are carried to the motor controller and how to mange them.

**Comment by melidron** [comment]
> Yes, the "shorting" diagram indicates that you don't need PWM to do speed control just a simple switch. For bare bones open loop speed control, maybe you don't even need a microcontroller. If you have the make and model number of the treadmill, perhaps we can find some more information on the signals that are carried to the motor controller and how to mange them.

i have found  the service manual of a treadmill that has almost same treadmill motor board... I wish u can understand what it needs to make it work again :)

[z700a82.pdf](https://github.com/user-attachments/files/17685323/z700a82.pdf)
